### ISSUE TITLE
Page content resizes with noticable delay when layout is heavy

### ISSUE THREAD

**Issue by evgenyzinoviev** [atom/electron#3615]
When you have quite heavy layout with many elements, page content resizes with a delay when you resizing a window.

Let's look at this. For the demo I've created a page with 1000 fixed elements.

This is what happens when I'm resizing an Electron window:
![resize-electron](https://cloud.githubusercontent.com/assets/6204306/11452713/1bba4318-95f2-11e5-8e15-b25591d9bd0b.gif)

I've also created a native cocoa app with a webview (which uses system Safari I guess) and it doesn't have this problem:
![resize-webview-safari](https://cloud.githubusercontent.com/assets/6204306/11452718/4c20e4c6-95f2-11e5-850d-f21e1193b7f8.gif)

So I'm looking for some solution. I'm ready to dig into Electron code, patch something, maybe do some dirty hack, there must be something! If it's at all possible, please give me some hint of how to fix it.

**Comment by etiktin** [comment]
Can you run that code in Chrome and check how it performs compared to Electron?

**Comment by bengotow** [comment]
What version of Electron are you using? There were some issues with resizing speed because Electron's windows were registering as "transparent regions" on Mac OS X, but I think that's been fixed in the latest builds.

**Comment by evgenyzinoviev** [comment]
@bengotow, Electron version is 0.35.2. The same happens on Windows.
@etiktin it's the same, I think it's because Chrome resizes everything asynchronously, after the window was resized. But maybe there is a way to make it synchronous? I'm not an expert of libchromiumcontent, but someone must know.

**Comment by DerNivel** [comment]
Yeah. The lag is really ugly if using a frameless window. A fix for this would be awesome.

**Comment by evgenyzinoviev** [comment]
And like I said, if someone give me a hint I can try to fix it myself. Maybe it's possible to do some hack, like wait for some 'resize-done' event from chromium and only after that actually resize a window... I don't know, there must be a way.

**Comment by raleksandar** [comment]
I can confirm this too, but just tested in Chrome and it's the same so I'm afraid there is not much we can do except wait for it to be fixed in Chromium :(

**Comment by zcbenz** [comment]
I don't think there is a way to make Chrome resize synchronously, the whole design of Chromium's rendering is to be asynchronous, and I'm pretty sure Chrome team won't bother adding it, so I'm closing this as won't fix. But if someone came up with a solution, I would be very happy to merge it.

**Comment by abhishiv** [comment]
Will this help - https://code.google.com/p/chromium/issues/detail?id=531831

**Comment by fuermosi777** [comment]
Do you have `window.on('resize', func)` or `remote.getCurrentWindow('resize')` event listeners setup?

**Comment by jbonnett92** [comment]
I had the same problem, I used the command `npm update electron-prebuilt -g` and I don't have the problem anymore. I think it's due to Chrome 55 that has just been released.

Make sure you update for the package you used (if you used any).

**Comment by pie6k** [comment]
Seems it's still an issue :< 

I was trying different 'hax'es like sending dimensions on 'will resize' to renderer and modifying container size manually etc. Nothing seems to help.

This is especially ugly when you resize the window by dragging top edge of it, then bottom part of the content is jumping quickly until I stop resizing.

**Comment by blucell** [comment]
Latest Chrome version now syncs window and render sizes.

https://user-images.githubusercontent.com/4368958/129436191-9fe2adf1-edd4-42d5-a27d-97b372dd0733.mov

Yet still, if relayout takes a lot of time, this split still occurs, especially for more complex layouts and because chrome is not optimizing for relayout performance.

**Comment by CryUshio** [comment]
> Latest Chrome version now syncs window and render sizes.

Reproduces with Electron 21.x. Any update?
